packages/clay-core/src/vertical-bar/Panel.tsx

```diff
@@ -52,32 +52,36 @@ export function Panel({children, keyValue = null, tabIndex}: Props) {
 		panelWidthMax,
 		panelWidthMin,
 		resize,
+		setPanelNext,
 	} = useContext(VerticalBarContext);
 	const {displayType} = useContext(ContentContext);
 
 	const nodeRef = useRef<HTMLDivElement | null>(null);
 
-	const previousActivePanelRef = useRef<React.Key | null>(null);
-
 	const {prefersReducedMotion} = useProvider();
 
 	const isFirst = useIsFirstRender();
 
+	const previousActivePanel = usePrevious(activePanel);
+	const previousPanelNext = usePrevious(panelNext);
+
 	const isPanelOpen = () => {
 		if (isFirst) {
 			return activePanel !== null;
-		} else if (previousActivePanelRef.current === activePanel) {
+		}
+
+		if (previousActivePanel === activePanel) {
 			return true;
-		} else if (activePanel === null) {
+		}
+
+		if (activePanel === null) {
```

Comment:
@veroglez I don't think we need this block, since we are returning false by default.

Reply:
Uppss! That slipped past me 😅